### ISSUE TITLE
Microsoft Teams, WinRAR, Skype unable to start when EP is registered as shell extension

### ISSUE THREAD

**Issue by PGomersall** [valinet/ExplorerPatcher#259]
If you enable the shell extension (22000.282.32.3) and have the full  MS Teams version installed and on the Taskbar then Teams breaks when you try to launch it from the Taskbar or Start. Basically once you enable the shell extension then clicking on Teams causes Microsoft Chat to try and launch but breaks. Unregistering the extension restores Teams to working order.

**Comment by valinet** [comment]
Hi

I have looked on it, I cannot seem to figure it out from a quick glance so to speak. It's pretty weird, even if I remove all the code in EP's DLL, it still hangs. Definitely a weird issue that requires more investigation.

But let me ask, why do you need this? The README only instructs you to drop `dxgi.dll` in `C:\Windows` and that is the recommended mode of operation and the one I test most frequently with and the one which you also reported works just fine, as it should. If you don't have much idea what this might be used for, then just ask first and then use it, if you still think it's worth it, especially since it is being labeled "experimental". It's a work in progress, and injecting all processes requires some care. And it's only useful for classic theme users at the moment, it doesn't enable much other functionality, so not something most users should do, again.

So, as I said in a couple of places, as a shell extension, this is still experimental, way less tested, not necessary for 99% of use cases, and recommended only for specific use cases, so please do not use this other than for testing (and hopefully contributing back some patches, as I am not really interested in working too much on this on my own, at least at the moment, since I do not use it at all) or some specific use case you should know about ahead of time.

Thanks for the report, I will look into it more when I have the opportunity; for the moment, I have no idea and not much time to investigate it too thoughtfully; maybe someone wants to pick it up.

**Comment by Biliskn3r** [comment]
// And it's only useful for classic theme users at the moment, it doesn't enable much other functionality, //

That's good to know, I was wondering what it'd be useful for. Cheers.

**Comment by valinet** [comment]
Yeah, well, it's no secret, I just did not take the time to update the README, but nevertheless, if one's in doubt of anything, better ask first rather than regret later, if the case, so to speak. It's not the case here, of course, as this software is harmless (and open, anyone can inspect it etc etc), but in general, please don't run things that you don't know what they do, at least in broad terms... And here specifically, I usually answer questions fairly quickly, there are also other users, the Discussions tab, so questions are (usually and mostly =))) fine.

**Comment by Anixx** [comment]
> That's good to know, I was wondering what it'd be useful for. Cheers.

Shell extension is needed s to affect the file open/save dialogs: if you want to change the context menus there or remove the navbar.

**Comment by dlx-GH** [comment]
I've installed the 22000.282.32.4 prerelease build- not shell extension- and my Teams starts but hangs when the UI appears.

Should I uninstall and try the 22000.282.31.4 release? Can I just exchange the DLL and reboot?

BTW- thanks sooo much for this awesome work!  Like.. WTH is MS thinkin??  
I'll have to buy you a beer ;)

**Comment by valinet** [comment]
When installed in `C:\Windows`, EP does not affect other applications, it only gets loaded in Explorer. I have tested it yesterday, Teams loaded just fine when the DLL was just in `C:\Windows`. Make sure you have deregistered the shell extension as well, that's how you "uninstall" it: `regsvr32 /u C:\Windows\dxgi.dll`.

**Comment by valinet** [comment]
I seem to have fixed it:

![Screenshot (2)](https://user-images.githubusercontent.com/6503598/141235563-ab4d8997-1f7c-4fde-ae5d-d17b310a570c.png)

I have tested it with 32-bit Paint from Windows Vista, Skype 8.78.0.159, Microsoft Teams 1.4.00.26376, Discord 104276, WinRAR 6.02 (64-bit) and it seemed to work, the shell extension works fine and enables a few functionalities for Open/Save file dialogs, like hiding the navigation bar, hiding the search bar, plus fixes the context menus when using the classic theme. There were a few issues and bugs that I had to iron out, and I think this is on the right track, it seems to get ready for a stable long term release.

Look out for build 22000.318.36.0 soon.

Thanks for the reports.
